### PR TITLE
test-execute: add test for that DynamicUser= migrates StateDirectory=

### PR DESCRIPTION
**Pull request by yuwata**
This cleans-up test-execute and adds a test for 949befd3f09e8c06a908ec99efd241666c21d944 (#7507).

### CODE REVIEW

File: src/test/test-execute.c [hunk 1]
```diff
@@ -139,6 +139,8 @@ static void test_exec_personality(Manager *m) {
 
 #elif defined(__i386__)
         test(m, "exec-personality-x86.service", 0, CLD_EXITED);
+#else
+        log_notine("Unknown architecture, skipping %s", __func__);
```
Comment (keszybz) [inline src/test/test-execute.c]: Typo here.

File: src/test/test-execute.c [hunk 2]
```diff
@@ -339,6 +339,13 @@ static void test_exec_dynamicuser(Manager *m) {
         test(m, "exec-dynamicuser-fixeduser-one-supplementarygroup.service", 0, CLD_EXITED);
         test(m, "exec-dynamicuser-supplementarygroups.service", 0, CLD_EXITED);
         test(m, "exec-dynamicuser-statedir.service", 0, CLD_EXITED);
+
+        test(m, "exec-dynamicuser-statedir-migrate-step1.service", 0, CLD_EXITED);
+        test(m, "exec-dynamicuser-statedir-migrate-step2.service", 0, CLD_EXITED);
+        (void) rm_rf("/var/lib/test-dynamicuser-migrate", REMOVE_ROOT|REMOVE_PHYSICAL);
+        (void) rm_rf("/var/lib/test-dynamicuser-migrate2", REMOVE_ROOT|REMOVE_PHYSICAL);
+        (void) rm_rf("/var/lib/private/test-dynamicuser-migrate", REMOVE_ROOT|REMOVE_PHYSICAL);
+        (void) rm_rf("/var/lib/private/test-dynamicuser-migrate2", REMOVE_ROOT|REMOVE_PHYSICAL);
```
Comment (keszybz) [inline src/test/test-execute.c:350]: It's a bit scary to play with the real filesystem like this... OTOH, the name is pretty unique, so this should be fine.

Comment (yuwata) [reply src/test/test-execute.c:350]: Yeah, I agree with you...

File: src/test/test-execute.c (hunk 1 above)
Comment (yuwata) [reply src/test/test-execute.c]: Thanks. Fixed.

**Comment by yuwata** [comment]
@keszybz I've force-pushed an updated version, which fixes a typo. Thank you.

File: src/test/test-execute.c [hunk 3]
```diff
@@ -139,6 +139,8 @@ static void test_exec_personality(Manager *m) {
 
 #elif defined(__i386__)
         test(m, "exec-personality-x86.service", 0, CLD_EXITED);
+#else
+        log_notine("Unknown personality, skipping %s", __func__);
```
Comment (keszybz) [inline src/test/test-execute.c]: Typo still there!

Comment (yuwata) [reply src/test/test-execute.c]: Ughhh... Sorry, updated.

File: src/test/test-execute.c [hunk 4]
```diff
@@ -233,67 +227,77 @@ static void test_exec_readonlypaths(Manager *m) {
 static void test_exec_readwritepaths(Manager *m) {
 
         if (path_is_read_only_fs("/") > 0)
+                log_notice("Root directory is readonly, skipping %s", __func__);
                 return;
```
Comment (poettering) [inline src/test/test-execute.c]: this needs to be enclosed in {} now, since it contains two lines now

File: src/test/test-execute.c [hunk 5]
```diff
@@ -233,67 +227,77 @@ static void test_exec_readonlypaths(Manager *m) {
...
-                log_notice("testing without inaccessible, skipping %s", __func__);
+                log_notice("Testing without inaccessible, skipping %s", __func__);
                 return;
         }
 
         test(m, "exec-inaccessiblepaths-proc.service", 0, CLD_EXITED);
+
+        if (path_is_read_only_fs("/") > 0)
+                log_notice("Root directory is readonly, skipping remaining tests in %s", __func__);
+                return;
```
Comment (poettering) [inline src/test/test-execute.c]: as above

File: src/test/test-execute.c (hunk 4 above)
Comment (yuwata) [reply src/test/test-execute.c]: Ouch! Updated.

**Comment by yuwata** [comment]
@poettering Thank you for the review. I've force-pushed new version. PTAL.

**Pull request merged**